src/pretix/base/models/discount.py

```diff
@@ -286,11 +305,25 @@ def apply(self, positions: Dict[int, Tuple[int, Optional[int], Decimal, bool, De
             )
         ]
 
+        if self.benefit_same_products:
+            benefit_candidates = list(condition_candidates)
+        else:
+            benefit_products = {p.pk for p in self.benefit_limit_products.all()}
+            benefit_candidates = [
+                idx
+                for idx, (item_id, subevent_id, line_price_gross, is_addon_to, voucher_discount) in positions.items()
+                if (
+                    item_id in benefit_products and
+                    (self.condition_apply_to_addons or not is_addon_to) and
+                    (not self.condition_ignore_voucher_discounted or voucher_discount is None or voucher_discount == Decimal('0.00'))
```

Comment:
Do we need extra checkboxes for applying benefit to addons and ignoring voucher-discounted? As both checkboxes currently are located in the condition fieldset, maybe one can overlook, that this also applies to the benefits? Not sure though. Or move both checkboxes to the top? Might also need a rename in the code then as it is not `condition_XXX` but also `benefit_XXX`, so maybe rename to `ignore_voucher_discounted` and `apply_to_addons`? On the other hand probably to much hassle.